app/src/main/java/org/listenbrainz/android/viewmodel/SocialViewModel.kt

```diff
@@ -171,11 +184,17 @@ class SocialViewModel @Inject constructor(
         }
     }
 
-   fun getFollowers(username : String) : Resource<SocialData> = runBlocking {
-        val deferredResult = async{
-            repository.getFollowers(username)
-        }
-       return@runBlocking deferredResult.await()
+   fun getFollowers() {
```

Comment:
We need this function to be synchronous and have UI do state management on its own since this function won't be used everytime this View model is created.
```suggestion
   suspend fun getFollowers(): Resource<SocialData> {
           val username = appPreferences.getUsername()
           return repository.getFollowers(username).also {
               if(it.status == Resource.Status.FAILED){
                   emitError(result.error)
               } 
           } 
    }
```
Some of the usages of `appPreferences.getUsername()` are old and use `withContext(ioDispatcher)` but the approach is actually redundant.